clusterloader2/pkg/measurement/common/simple/service_creation_latency.go

```diff
@@ -0,0 +1,538 @@
+		createToAvailableLatency = append(createToAvailableLatency, serviceCreationLatencyData{
+			Key:     key,
+			Latency: ct.available.Sub(ct.creation),
+		})
+	}
+
+	sort.Sort(serviceCreationLatencySlice(createToAssignedLatency))
+	sort.Sort(serviceCreationLatencySlice(assignedToAvailableLatency))
+	sort.Sort(serviceCreationLatencySlice(createToAvailableLatency))
+
+	latency := &serviceCreationLatency{
+		CreateToAssignedLatency:    extractLatencyMetrics(createToAssignedLatency),
+		AssignedToAvailableLatency: extractLatencyMetrics(assignedToAvailableLatency),
+		CreateToAvailableLatency:   extractLatencyMetrics(createToAvailableLatency),
+	}
+	content, err := util.PrettyPrintJSON(podStartupLatencyToPerfData(latency))
+	if err != nil {
+		return nil, err
+	}
+	summary := measurement.CreateSummary(serviceCreationLatencyName, "json", content)
+	return []measurement.Summary{summary}, err
+}
+
+type serviceCreationLatencyData struct {
+	Key     string
+	Latency time.Duration
+}
+
+type serviceCreationLatencySlice []serviceCreationLatencyData
+
+func (a serviceCreationLatencySlice) Len() int           { return len(a) }
+func (a serviceCreationLatencySlice) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a serviceCreationLatencySlice) Less(i, j int) bool { return a[i].Latency < a[j].Latency }
+
+func extractLatencyMetrics(latencies []serviceCreationLatencyData) measurementutil.LatencyMetric {
+	length := len(latencies)
+	if length == 0 {
+		// Ideally we can return LatencyMetric with some NaN/incorrect values,
+		// but 0 is the best we can get for time.Duration type.
+		return measurementutil.LatencyMetric{Perc50: 0, Perc90: 0, Perc99: 0}
+	}
+	perc50 := latencies[int(math.Ceil(float64(length*50)/100))-1].Latency
+	perc90 := latencies[int(math.Ceil(float64(length*90)/100))-1].Latency
+	perc99 := latencies[int(math.Ceil(float64(length*99)/100))-1].Latency
+	return measurementutil.LatencyMetric{Perc50: perc50, Perc90: perc90, Perc99: perc99}
+}
+
+type serviceCreationLatency struct {
+	CreateToAssignedLatency    measurementutil.LatencyMetric `json:"createToAssignedLatency"`
+	AssignedToAvailableLatency measurementutil.LatencyMetric `json:"assignedToAvailableLatency"`
+	CreateToAvailableLatency   measurementutil.LatencyMetric `json:"createToAvailableLatency"`
+}
+
+func podStartupLatencyToPerfData(latency *serviceCreationLatency) *measurementutil.PerfData {
+	const nsToMsRatio = 1000000.0
+	perfData := &measurementutil.PerfData{Version: "1.0"}
+	perfData.DataItems = append(perfData.DataItems, latency.CreateToAssignedLatency.ToPerfData("create_to_assigned", nsToMsRatio))
+	perfData.DataItems = append(perfData.DataItems, latency.AssignedToAvailableLatency.ToPerfData("assigned_to_available", nsToMsRatio))
+	perfData.DataItems = append(perfData.DataItems, latency.CreateToAvailableLatency.ToPerfData("create_to_available", nsToMsRatio))
+	return perfData
+}
+
+func (s *serviceCreationLatencyMeasurement) newInformer() (cache.SharedInformer, error) {
```

Comment:
It's pretty clear already that we may need different of those.
Instead of adding it here, you should create a file (probably something like pkg/framework/informers.go) and add helpers to create informers there (also move the existing one for pods) - majority of code is shared between those and can be extracted.

Reply:
Done.